mapper/tedge_mapper/src/sm_c8y_mapper/smartrest_serializer.rs

```diff
@@ -0,0 +1,541 @@
+use crate::sm_c8y_mapper::error::SmartRestSerializerError;
+use csv::{QuoteStyle, WriterBuilder};
+use json_sm::{
+    SoftwareListResponse, SoftwareOperationStatus, SoftwareType, SoftwareUpdateResponse,
+    SoftwareVersion,
+};
+use serde::{Deserialize, Serialize, Serializer};
+
+type SmartRest = String;
+
+#[derive(Debug)]
+pub(crate) enum CumulocitySupportedOperations {
+    C8ySoftwareUpdate,
+}
+
+impl From<CumulocitySupportedOperations> for &'static str {
+    fn from(op: CumulocitySupportedOperations) -> Self {
+        match op {
+            CumulocitySupportedOperations::C8ySoftwareUpdate => "c8y_SoftwareUpdate",
+        }
+    }
+}
+
+pub(crate) trait SmartRestSerializer<'a>
+where
+    Self: Serialize,
+{
+    fn to_smartrest(&self) -> Result<SmartRest, SmartRestSerializerError> {
+        serialize_smartrest(self)
+    }
+}
+
+#[derive(Debug, Deserialize, Serialize, PartialEq)]
+pub(crate) struct SmartRestSetSupportedOperations {
+    pub message_id: &'static str,
```

Comment:
Why is `message_id` an &str and not a `u16` or something numeric type? Is it because the CSV parser can't distinguish between numbers and strings while parsing a CSV string?